syntax = "proto3";
import "google/protobuf/empty.proto";
option go_package = ".;proto";

service User{
  rpc GetUserList(PageInfo) returns (UserListResponse); //用户列表
  rpc GetUserByMobile(MobileRequest) returns (UserInfoResponse); // 通过mobile查询用户
  rpc GetUserById(IdRequest) returns (UserInfoResponse); // 通过Id查询用户
  rpc CreateUser(CreateUserInfo) returns (UserInfoResponse); // 创建用户
  rpc UpdateUser(UpdateUserInfo) returns (google.protobuf.Empty); // 更新用户信息
  rpc CheckPassword(PasswordCheckInfo) returns (CheckPasswordResponse); // 校验密码
}

message PasswordCheckInfo{
  string password = 1;
  string encryptedPassword = 2;
}

message CheckPasswordResponse{
  bool  success = 1;
}
message PageInfo{
  uint32 page = 1;
  uint32 size = 2;
}
message MobileRequest{
  string mobile = 1;
}
message IdRequest{
  int32 id = 1;
}

message CreateUserInfo{
  string nickname = 1;
  string password = 2;
  string mobile = 3;

}

message UpdateUserInfo{
  int32 id = 1;
  string nickname = 2;
  string gender = 3;
  uint64 birthday = 4;

}

message UserListResponse{
  int32 total = 1;
  repeated UserInfoResponse data = 2;
}

message UserInfoResponse{
  int32 id = 1;
  string nickname = 2;
  string password = 3;
  string mobile = 4;
  uint64 birthday = 5;
  string gender = 6;
  string role = 7;
}